filecoindot-rpc/src/lib.rs

```diff
@@ -17,7 +17,7 @@ mod result;
 pub trait FilecoindotApi {
     /// set filecoin rpc endpoint for filecoindot
     #[rpc(name = "filecoindot_setRpcEndpoint")]
-    fn set_rpc_endpoint(&self, url: String) -> Result<()>;
+    fn set_rpc_endpoint(&self, urls: String) -> Result<()>;
```

Comment:
I think this should be `fn set_rpc_endpoint(&self, urls: &[String]) -> Result<()>;` to have each urls in a separate string, rather than splitting them in commas later.

Reply:
there are some encoding issues serialize and save `Vec<String>` into the offchain db, losing bytes, I'll retry this and try to find the root case